ls1intum/Artemis
# Add Edit, Participations and Configure Grading Buttons

### Checklist

### Motivation and Context
The instructor actions in the exercise normal course view are very useful, and already offer shortcuts to some pages without loading the complete course management page each time and searching for the correct exercise.

In this PR, we want to add some more buttons that are frequently used and are "missing" in the current version:
- **"Edit"-Button**
  _Use Case: an instructor reads or looks up the exercise in the course view and found mistakes in the problem statement (e.g. spelling errors). The edit button allows to quickly edit the problem statement and make corrections. Exercise release date, due date and so on can also be adjusted quickly._
- **"Participations"-Button** (for programming exercises only)
  _Use Case: allows quickly checking if and how students participated, and investigate problems. An instructor could also trigger a rebuild for a particular student participation if necessary. There are many scenarios related to that based on our (PGdP-ÜL) experience. This would make it just a bit more pleasant._
- **"Configure Grading"-Button** (for programming exercises only)
  _Use case: quick navigation between exercise editing, scores, participations and grading. Instructors can adjust grading based on the scores and automatic feedback that students got, or look up the problem statement again and adjust the test cases (or reverse). This together with the others is estimated to "increase the quality of life" as an instructor of a course with a lot of programming exercises._

In summary, most points just improve navigation centered around issues related to a single exercise at a time. The main reason this is required is that neither #611 is implemented nor does the course management load quickly while restoring the old view (exercise sorting, scroll-position), making it necessary find the exercise for each switch "from scratch".

#### Alternatives
- Restore the course management quickly and the way it has been left when navigating back in the browser.
- Resolve issue #611, although this might still be slower due to full page reload.
- Offer some kind of quick navigation / quick search for exercises, this would fit really well to #2097. But this could also be some sort of extra page just for quick navigation and re-routing related to a single exercise.

### Description
1. Added an edit button for all exercise types and participations and configure
grading only for programming exercises because they are mostly needed
there.
2. Changed all buttons to the design from the course exercise management page.
3. Renamed "Configure Grading" buttons to "Grading" for programming exercises.

### Steps for Testing
_Short version: test that all "Instructor actions" buttons work._

0. You need to be instructor in a course with at least one of each exercise type.
1. Check that edit opens the edit page for each exercise type.
    For quiz exercise that are already over, you should just see that editing is not possible anymore.
2. All exercises should still have a "Scores" button. Check that the "Scores" button leads to the exercise's scores.
3. Check that only for programming exercises the "Participations" and "Configure Grading" are displayed.
    For programming exercises, the "Participations" should redirect to the exercise's participations and "Configure Grading" to the grading configuration page of the exercise.
4. Modelling exercises have "Submissions" and "Statistics" buttons and text exercises have the "Submissions" button as well.
   Check that the "Submissions" button leads to the submissions for both types and that the "Statistics" page can be opened in case of modelling exercises.

### Screenshots


- **Modelling Exercise Example:**
   ![image](https://user-images.githubusercontent.com/11130248/95689602-34389480-0c12-11eb-9a1d-c81844763ea8.png)
- **Programming Exercise Example:**
   ![image](https://user-images.githubusercontent.com/11130248/95690026-2d5f5100-0c15-11eb-90fd-dfd0a27006c8.png)
